src/test/java/net/kaczmarzyk/spring/data/jpa/domain/InTheFutureTest.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2014-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.domain;
+
+import net.kaczmarzyk.spring.data.jpa.ComparableTestBase;
+import net.kaczmarzyk.spring.data.jpa.Customer;
+import net.kaczmarzyk.spring.data.jpa.utils.Converter;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.time.OffsetDateTime;
+
+import static net.kaczmarzyk.spring.data.jpa.CustomerBuilder.customer;
+
+public class InTheFutureTest extends ComparableTestBase {
+
+	private static final OffsetDateTime FUTURE_DATE_TIME = OffsetDateTime.now().plusDays(10);
+	private static final OffsetDateTime PAST_DATE_TIME = OffsetDateTime.now().minusMonths(10);
+
+	@Before
+	public void initData() {
+		homerSimpson = customer("Homer", "Simpson")
+				.registrationDate(FUTURE_DATE_TIME.getYear(), FUTURE_DATE_TIME.getMonth().getValue(), FUTURE_DATE_TIME.getDayOfMonth())
+				.nextSpecialOffer(FUTURE_DATE_TIME)
+				.birthDate(PAST_DATE_TIME.toLocalDate())
+				.lastOrderTime(FUTURE_DATE_TIME.toLocalDateTime())
+				.build(em);
+		margeSimpson = customer("Marge", "Simpson")
+				.registrationDate(FUTURE_DATE_TIME.getYear(), FUTURE_DATE_TIME.getMonth().getValue() + 2, FUTURE_DATE_TIME.getDayOfMonth())
+				.nextSpecialOffer(PAST_DATE_TIME)
+				.birthDate(FUTURE_DATE_TIME.toLocalDate())
+				.lastOrderTime(FUTURE_DATE_TIME.toLocalDateTime().plusMonths(2))
+				.build(em);
+		moeSzyslak = customer("Moe", "Szyslak")
+				.registrationDate(PAST_DATE_TIME.getYear(), PAST_DATE_TIME.getMonth().getValue(), PAST_DATE_TIME.getDayOfMonth())
+				.nextSpecialOffer(FUTURE_DATE_TIME.plusDays(30))
+				.birthDate(FUTURE_DATE_TIME.toLocalDate().plusDays(10))
+				.lastOrderTime(PAST_DATE_TIME.toLocalDateTime())
+				.build(em);
```

Comment:
Please verify is there any way to increasing the test data readability.

By adding comment like:
```
/*
*          | registration | next special offer | birth date | last order time
* ------  | ------------ | ------------------ | ---------- | ----------------
* HOMER   |    FUTURE    |       FUTURE       |    PAST    |    FUTURE
* MARGE   |    FUTURE    |       PAST         |   FUTURE   |    FUTURE
* MOE        |    PAST      |      FUTURE        |   FUTURE   |     PAST
*/
```
or by extending the customer builder:
```
		customer("Homer", "Simpson").with(
						REGISTRATION_DATE, IN_THE_FUTURE
						NEXT_SPECIAL_OFFER, IN_THE_FUTURE
						BIRTH_DATE, IN_THE_PAST,
						LAST_ORDER_TIME, FUTURE_DATE_TIME
				).build(em);
```
or
```
		customercustomer("Homer", "Simpson")
                  .withAllDatesInTheFuture()
                  .withBirthDate(IN_THE_PAST).build(em);
```